packages/toolbox-core/src/toolbox_core/tool.py

```diff
@@ -330,3 +326,12 @@ def params_to_pydantic_model(
             ),
         )
     return create_model(tool_name, **field_definitions)
+
+
+async def get_value(func: Callable[[], Any]) -> Any:
+    """Asynchronously or synchronously gets the value from a callable."""
+    if asyncio.iscoroutinefunction(func):
+        return await func()
+    elif callable(func):
```

Comment:
There's no default case in this situation -- what do we do if it's not a await-able or callable?

Reply:
I think that means token getters can now be Callables, Awaitables, or just Any. Should we update the type hint for the function?